lib/galaxy/datatypes/metacyto.py

```diff
@@ -0,0 +1,108 @@
+# -*- coding: utf-8 -*-
+######################################################################
+#                  Copyright (c) 2016 Northrop Grumman.
+#                          All rights reserved.
+######################################################################
+
+"""
+MetaCyto analysis datatypes.
+"""
+
+import logging
+
+from galaxy.datatypes.tabular import Tabular
+from . import data
+
+log = logging.getLogger(__name__)
+
+
+class mStats(Tabular):
+    """Class describing the table of cluster statistics output from MetaCyto"""
+    file_ext = "metacyto_stats.txt"
+
+    def set_peek(self, dataset, is_multi_byte=False):
+        if not dataset.dataset.purged:
+            dataset.peek = "MetaCyto Cluster Stats file"
+            dataset.blurb = data.nice_size(dataset.get_size())
+        else:
+            dataset.peek = 'file does not exist'
+            dataset.blurb = 'file purged from disk'
+
+    def display_peek(self, dataset):
+        try:
+            return dataset.peek
+        except Exception:
+            return "MetaCyto Cluster Stats file (%s)" % (data.nice_size(dataset.get_size()))
+
+    def sniff(self, filename):
+        """Quick test on file headings"""
+        with open(filename, "r") as f:
+            # last one == fraction
+            headers = f.readline().strip().split("\t")
+            if headers[-1] != "fraction":
+                return False
+            hdrs = ["fcs_files", "cluster_id", "label", "fcs_names"]
+            for h in hdrs:
```

Comment:
Maybe use Set intersection?

Reply:
Thanks, I have changed all of these for set intersections as suggested.